components/channel_members_rhs/channel_members_rhs.tsx

```diff
@@ -88,10 +106,32 @@ export default function ChannelMembersRHS({channel, searchTerms, membersCount, c
 
         actions.setChannelMembersRhsSearchTerm('');
         setEditing(false);
-        actions.loadProfilesAndReloadChannelMembers(channel.id);
+        setPage(0);
+        setLoadingMore(false);
+
+        actions.loadProfilesAndReloadChannelMembers(0, USERS_PER_PAGE, channel.id);
         actions.loadMyChannelMemberAndRole(channel.id);
     }, [channel.id, channel.type]);
 
+    const searchDebounced = debounce(
```

Comment:
This is going to get redefined every time this re-renders, so you'll lose the state tracking the debouncing. I'm not sure exactly how to make this work in a hook, but maybe something like wrapping it in a `useMemo` would work to fix that?